oak_tdx_guest/src/tdcall.rs

```diff
@@ -0,0 +1,209 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Rust implementation of the TDX TDCALL instruction.
+
+use bitflags::bitflags;
+use core::{arch::asm, mem::size_of};
+
+/// The result from an instruction that indicates success.
+const SUCCESS: u64 = 0;
+
+bitflags! {
+    /// Attributes of a TD.
+    ///
+    /// See section 18.2.1 of <https://www.intel.com/content/dam/develop/external/us/en/documents/intel-tdx-module-1eas.pdf>
```

Comment:
Incidentally, it always scares me to refer to such URLs in technical documentation, it feels they may disappear / change any time. Perhaps a good use case for ent / static space (e.g. [content-addressed link to that PDF](https://static.space/8tXgarnYWVAhmV5gYDtTZMmnHvQAEVutxbzdnMPLosxMvhS2HPhbTK7jP4No7h8AUdoFz1EjddGbBb3NcaaCjuv4MJ))

Reply:
I have mixed feelings about it. I agree that URLs changing is a problem. I recently had to fix broken URLs for the AMD GHCB specification. The downside of using a content-addressed link to a PDF is that a reviewer of this code does not know the source of the content. So they don't know whether it is the exact specification as published by Intel, or some modified version (e.g. perhaps with embedded malware). I would be sceptical of downloading PDFs from content-addressed stores on untrusted repositories.

For now I have added a text description of the link so that someone can search for the document if the URL changes.